src/driver/implementation/imap/imapdriver_tools.c

```diff
@@ -1716,7 +1716,7 @@ static int imap_mailbox_list_to_group(clist * imap_mb_list, clistiter ** iter,
   imap_mailbox_listiter = * iter;
 
   imap_addr = clist_content(imap_mailbox_listiter);
-  if (imap_addr->ad_mailbox_name == NULL) {
+  if (imap_addr != NULL && imap_addr->ad_mailbox_name == NULL) {
```

Comment:
I'm pretty sure having imap_addr == NULL is not a valid use case.
(and it would crash on lines below)

So, there's two options:
Either we test at the beginning of the function and return MAIL_ERROR_INVAL or  maybe the code is fine after all and imap_mailbox_list_to_group() is used only with the content of iter not NULL.

Reply:
I think you right, better test is:
if (imap_addr == NULL || imap_addr->ad_mailbox_name == NULL) {
What do you think?